src/Tasks/MigrationTaskStep2.php

```diff
@@ -0,0 +1,321 @@
+                '{FileTable}' => $schema->tableName(File::class),
+                '{VersionSuffix}' => $tableSuffix,
+                '{RootRelationName}' => $do->ClassName,
+                '{ID}' => $do->ID,
+            ]
+        ));
+
+        $newData = [];
+        foreach($oldData as $item) {
+            $newItem = $this->getAsset($item['FilePublicID']);
+            $newItem->title = $item['Caption'];
+            $newItem->description = $item['AltText'];
+            $newItem->gravity = $item['Focus'];
+            array_push($newData, $newItem);
+        }
+
+        $sql = sprintf(
+            'UPDATE "%s" SET "%s" = \'%s\' WHERE "ID" = %d',
+            $tableName . $tableSuffix,
+            $fieldName,
+            DB::get_conn()->escapeString(json_encode($newData)),
+            $do->ID
+        );
+
+        DB::query($sql);
+    }
+
+    /**
+     * @param DataObject $do
+     * @param string $fieldName
+     * @param string $tableSuffix
+     *
+     * @return void(0)
+     */
+    private function handleCloudinaryImage(DataObject $do, string $fieldName, string $tableSuffix)
+    {
+        $schema = DataObject::getSchema();
+
+        $tableName = $schema->tableForField($do->ClassName, $fieldName) . $tableSuffix;
+
+        $sql = strtr(
+            static::config()->get('has_one_sql_template'),
+            [
+                '{LinkTable}' => $schema->tableName(ImageLink::class),
+                '{FileTable}' => $schema->tableName(File::class) . $tableSuffix,
+                '{FieldName}' => $fieldName,
+                '{TableName}' => $tableName,
+                '{ID}' => $do->ID,
+            ]
+        );
+
+        $data = DB::query($sql)->first();
+        if (!$data || !array_key_exists('FilePublicID', $data) || !$data['FilePublicID']) {
+            return;
+        }
+        $newData = $this->getAsset($data['FilePublicID']);
+
+        if (!$newData) {
+            return;
+        }
+        $newData->title = $data['Caption'];
+        $newData->description = $data['AltText'];
+        $newData->gravity = $data['Focus'];
+
+        $sql = sprintf(
+            'UPDATE "%s" SET "%s" = \'%s\' WHERE "ID" = %d',
+            $tableName,
+            $fieldName,
+            DB::get_conn()->escapeString(json_encode($newData)),
+            $do->ID
+        );
+
+        DB::query($sql);
+    }
+
+    /**
+     * @uses API::resource()
+     */
+    private function getAsset($publicId)
+    {
+        $cacheKey = md5($publicId);
+
+        if ($this->cache->has($cacheKey)) {
+            echo '.';   // Just so we see progress when large number of cached responses
+            return $this->cache->get($cacheKey);
+        }
+
+        try {
+            $request = new HTTPRequest('GET', '/', [
+                'public_id' => $publicId,
+                'resource_type' => 'image'
+            ]);
+
+            var_dump(sprintf('Requesting [%d]: %s', ++$this->requestCount, $publicId));
```

Comment:
Do we have to use var_dump for these?

Reply:
Replaced with `static::output()` (new static function)